### DISCUSSION TITLE
UNPROTECTED SSL FILE

### DISCUSSION THREAD

**Discussion by haley0918** [Chia-Network/chia-blockchain#8429] [Support]
I'm running farmer on Pi4 with Ubuntu 20.04.3 LTS headless.
Upgrading to 1.2.5 and above gave me this warning with almost every chia command.

> (venv) ubuntu@ubuntu:~/chia-blockchain$ chia version
> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ 
> @             WARNING: UNPROTECTED SSL FILE!              @
> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ 
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/private_farmer.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/public_farmer.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/private_full_node.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.crt' are too open. Expected 0644 
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/harvester/private_harvester.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.crt' are too open. Expected 0644 
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/private_timelord.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/public_timelord.crt' are too open. Expected 0644                                                         
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/private_wallet.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/public_wallet.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/private_farmer.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/main/config/ssl/farmer/public_farmer.key' are too open. Expected 0600 
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/private_full_node.key' are too open. Expected 0600                                                      
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.key' are too open. Expected 0600                                                       
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.key' are too open. Expected 0600           
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.key' are too open. Expected 0600        
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/harvester/private_harvester.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/private_timelord.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/public_timelord.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/private_wallet.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/public_wallet.key' are too open. Expected 0600
> One or more SSL files were found with permission issues.
> Run `chia init --fix-ssl-permissions` to fix issues.
> 1.2.6

Executing --fix-ssl-permissions doesn't solve it.

> (venv) ubuntu@ubuntu:~/chia-blockchain$ chia init --fix-ssl-permissions
> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ @             WARNING: UNPROTECTED SSL FILE!              @
> @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.crt' are too open. Expected 0644          
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/private_farmer.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/public_farmer.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/private_full_node.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/harvester/private_harvester.crt' are too open. Expected 0644                                                      
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.crt' are too open. Expected 0644        
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/private_timelord.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/public_timelord.crt' are too open. Expected 0
> 644                                                         
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/private_wallet.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/public_wallet.crt' are too open. Expected 0644
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.key' are too open. Expected 0600           
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/private_farmer.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/public_farmer.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/private_full_node.key' are too open. Expected 0600                                                      
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/harvester/private_harvester.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.key' are too open. Expected 0600        
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/private_timelord.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/public_timelord.key' are too open. Expected 0600                                                         
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/private_wallet.key' are too open. Expected 0600
> Permissions 0777 for '/media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/public_wallet.key' are too open. Expected 0600
> One or more SSL files were found with permission issues.    Run `chia init --fix-ssl-permissions` to fix issues.
> Chia directory /media/ubuntu/BUP0/.chia/mainnet             
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.crt                        
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.crt            
>  Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/private_farmer.crt             
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/public_farmer.crt              > Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/private_full_node.crt       
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.crt        
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/harvester/private_harvester.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/private_timelord.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/public_timelord.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/private_wallet.crt
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/public_wallet.crt              > Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.key             
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/private_farmer.key             
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/farmer/public_farmer.key              > Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/private_full_node.key       
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.key        
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/chia_ca.key                        
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/harvester/private_harvester.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/full_node/public_full_node.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/ca/private_ca.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/private_timelord.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/timelord/public_timelord.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/daemon/private_daemon.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/private_wallet.key
> Attempting to set permissions 0777 on /media/ubuntu/BUP0/.chia/mainnet/config/ssl/wallet/public_wallet.key
> Finished updating SSL file permissions
> /media/ubuntu/BUP0/.chia/mainnet already exists, no migration action taken

**Comment by DArKer08** [comment]
Are you make symbolic link to an external NTFS HDD like me?

**Reply by DArKer08** [reply]
edit /etc/fstab mount permissions option as
`UUID=xxx /media/ubuntu ntfs-3g permissions,nofail`
and `chia init --fix-ssl-permissions` works fine.

**Comment by haley0918** [comment]
Yes I made symbolic link of ".chia" folder to external NTFS HDD

Unfortunately I mounted it on startup using udisksctl and it doesn't allow permissions in it's mount options At least not with udisks 2.8.4 in Ubuntu 20.04.

**Comment by ethanthekiwi** [comment]
Doesn't help OP, but for anyone else who comes across this. I saw this error in `~/.chia/mainnet/log` after upgrading to 1.2.7. 
`2021-09-21T05:36:02.477 farmer chia.farmer.farmer         : ERROR    Permissions 0664 for '/home/user/chia-blockchain/mozilla-ca/cacert.pem' are too open. Expected 0644`

I was able to fix it with the fix permissions command.
```
cd chia-blockchain
. ./activate
chia init --fix-ssl-permissions
```

**Reply by mauricecyril** [reply]
Hi is there a way to fix this on a drive that does not support permissions? Ie. Formated exfat on linux
